### PR TITLE
Feature/23754 pupil api tests

### PR DESCRIPTION
**Pull request by activemq**
Adding test for following scenarios

- Generating a pin > making a request to login via Pupil API > asserting response code + body

- Make a request with invalid credentials to Pupil API > asserting response code + body

### CODE REVIEW

**Comment by activemq** [comment]
@ reviewers - Next commit will be to remove all localhost:3003/auth and make it so it works with any environment

File: test/pupil/features/step_definitions/pupil_api.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  visit ENV['ADMIN_BASE_URL']+'/pupil-pin/generate-live-pins-list'
+  all('tbody tr').first.find('input[type="checkbox"]').click
+  find('.sticky-banner-wrapper #stickyConfirm').click
+  @school_password = all('.pin-content span:nth-child(2)').last.text
+  @pupil_pin = all('.pin-content span:nth-child(5)').last.text
+end
+
+Given(/^I make a request with valid credentials$/) do
+  step 'I have generated a pin via the admin app'
+  Timeout.timeout(60){sleep 1 until RequestHelper.auth(@school_password, @pupil_pin).code == 200}
```
Comment (GuyHarwood) [inline test/pupil/features/step_definitions/pupil_api.rb]: timeout should be configurable

File: test/pupil/features/step_definitions/pupil_api.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+end
+
+And(/^I should see a valid response$/) do
+  parsed_response = JSON.parse(@response.body)
+  expect(parsed_response['questions'].size).to eql 10
+  pupil_details = SqlDbHelper.find_pupil_via_pin(@pupil_pin)
+  expect(parsed_response['pupil']).to eql create_pupil_details_hash(pupil_details)
+  expect(parsed_response['school']).to eql create_school_details_hash(pupil_details['school_id'])
+  expect(parsed_response['config']).to eql create_config_details_hash
+  expect(parsed_response['tokens']['checkStarted']['url']).to eql 'https://strgt1dvmtcmq.queue.core.windows.net/check-started'
```
Comment (GuyHarwood) [inline test/pupil/features/step_definitions/pupil_api.rb]: URL should be extracted to parameter

**Comment by activemq** [comment]
@GuyHarwood please re review

File: test/pupil/features/step_definitions/pupil_api.rb [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  expect(parsed_response['tokens']['pupilPreferences']['token']).to_not be_nil
+  expect(parsed_response['tokens']['checkComplete']['url']).to include '/check-complete'
+  expect(parsed_response['tokens']['checkComplete']['token']).to_not be_nil
+  expect(parsed_response['tokens']['pupilFeedback']['url']).to include '/pupil-feedback'
+  expect(parsed_response['tokens']['pupilFeedback']['token']).to_not be_nil
+  expect(parsed_response['tokens']['jwt']['token']).to_not be_nil
+end
+
+Given(/^I make a request with invalid credentials$/) do
+  @response = RequestHelper.auth('abc12345', '9999')
```
Comment (GuyHarwood) [inline test/pupil/features/step_definitions/pupil_api.rb]: this _could_ eventually be a valid set of credentials when we add in test account functionality?

**Pull request merged**